pimd/pim_cmd.c

```diff
@@ -5560,72 +4260,54 @@ DEFUN (show_ip_pim_nexthop_lookup,
 {
 	struct prefix nht_p;
 	int result = 0;
-	struct in_addr src_addr, grp_addr;
-	struct in_addr vif_source;
-	const char *addr_str, *addr_str1;
+	pim_addr vif_source;
+	pim_addr src_addr, grp_addr;
 	struct prefix grp;
 	struct pim_nexthop nexthop;
 	char nexthop_addr_str[PREFIX_STRLEN];
-	char grp_str[PREFIX_STRLEN];
-	int idx = 2;
-	struct vrf *vrf = pim_cmd_lookup_vrf(vty, argv, argc, &idx);
+	struct vrf *v;
 
-	if (!vrf)
-		return CMD_WARNING;
+	v = vrf_lookup_by_name(vrf ? vrf : VRF_DEFAULT_NAME);
 
-	argv_find(argv, argc, "A.B.C.D", &idx);
-	addr_str = argv[idx]->arg;
-	result = inet_pton(AF_INET, addr_str, &src_addr);
-	if (result <= 0) {
-		vty_out(vty, "Bad unicast address %s: errno=%d: %s\n", addr_str,
-			errno, safe_strerror(errno));
+	if (!v)
 		return CMD_WARNING;
-	}
 
-	if (pim_is_group_224_4(src_addr)) {
+	if (pim_is_group_224_4(source)) {
 		vty_out(vty,
 			"Invalid argument. Expected Valid Source Address.\n");
 		return CMD_WARNING;
 	}
 
-	addr_str1 = argv[idx + 1]->arg;
-	result = inet_pton(AF_INET, addr_str1, &grp_addr);
-	if (result <= 0) {
-		vty_out(vty, "Bad unicast address %s: errno=%d: %s\n", addr_str,
-			errno, safe_strerror(errno));
-		return CMD_WARNING;
-	}
-
-	if (!pim_is_group_224_4(grp_addr)) {
+	if (!pim_is_group_224_4(group)) {
 		vty_out(vty,
 			"Invalid argument. Expected Valid Multicast Group Address.\n");
 		return CMD_WARNING;
 	}
 
-	if (!pim_rp_set_upstream_addr(vrf->info, &vif_source, src_addr,
-				      grp_addr))
+	if ((source.s_addr != INADDR_ANY) && (group.s_addr != INADDR_ANY)) {
```

Comment:
Why add this check on the addresses here? There wasn't one before?

Reply:
Done